network-config-analyzer/K8sYamlParser.py

```diff
@@ -0,0 +1,202 @@
+            self.syntax_error(f'invalid key "{key_label}", a label key name part must consist of alphanumeric '
+                              f'characters, "-", "_" or ".", and must start and end with an alphanumeric character',
+                              key_container)
+
+    def check_label_value_syntax(self, val, key, key_container):
+        """
+        checking validity of the label's value
+        :param string val : the value to be checked
+        :param string key: The key name which the value is assigned for
+        :param dict key_container : The label selector's part where the key: val appear
+        :return: None
+        """
+        if val is None:
+            self.syntax_error(f'value label of "{key}" can not be null', key_container)
+        if val:
+            if len(val) > 63:
+                self.syntax_error(f'invalid value in "{key}: {val}", a label value must be no more than 63 characters',
+                                  key_container)
+            pattern = r"(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?"
+            if re.fullmatch(pattern, val) is None:
+                self.syntax_error(f'invalid value in "{key}: {val}", value label must be an empty string or consist'
+                                  f' of alphanumeric characters, "-", "_" or ".", and must start and end with an '
+                                  f'alphanumeric character ', key_container)
+
+    def parse_label_selector_requirement(self, peer_container, namespace, requirement, namespace_selector):
+        """
+        Parse a LabelSelectorRequirement element
+        :param PeerContainer peer_container: The requirement will be evaluated against this set of peers
+        :param K8sNamespace namespace: The namespace in which the peers will be looked for
+        :param dict requirement: The element to parse
+        :param bool namespace_selector: Whether or not this is in the context of namespaceSelector
+        :return: A PeerSet containing the peers that satisfy the requirement
+        :rtype: Peer.PeerSet
+        """
+        self.check_fields_validity(requirement, 'LabelSelectorRequirement', {'key': [1, str], 'operator': [1, str],
+                                                                             'values': [0, list]},
+                                   {'operator': ['In', 'NotIn', 'Exists', 'DoesNotExist']})
+        key = requirement['key']
+        operator = requirement['operator']
+        self.check_label_key_syntax(key, requirement)
+        if operator in ['In', 'NotIn']:
+            values = requirement.get('values')
+            if not values:
+                self.syntax_error('A requirement with In/NotIn operator but without values', requirement)
+            action = self.FilterActionType.In if operator == 'In' else self.FilterActionType.NotIn
+            if namespace_selector:
+                return peer_container.get_namespace_pods_with_label(key, values, action)
```

Comment:
I'm seeing a lot of IDE warnings such as `Cannot find reference 'get_namespace_pods_with_label' in 'PeerContainer.py'` , because we are missing the import `from PeerContainer import PeerContainer`.
Adding the import would create some circular dependencies, but I think we can change the service yaml parse to inherit from the generic yaml parser instead of k8s yaml parser, so that we can add this import here.

Reply:
Changed the inheritance and united K8sYamlParser back with K8sPolicyYamlParser